examples/gameroom/daemon/src/rest_api/routes/submit.rs

```diff
@@ -64,39 +77,263 @@ pub fn submit_scabbard_payload(
+                RestApiResponseError::InternalError(format!("Failed to send request {}", err))
+            })
+            .and_then(move |mut resp| {
+                let body = match resp.body().wait() {
+                    Ok(b) => b,
+                    Err(err) => {
+                        return Either::B(future::err(RestApiResponseError::InternalError(
+                            format!("Failed to receive response body {}", err),
+                        )))
+                    }
+                };
+                match resp.status() {
+                    StatusCode::OK => {
+                        let batches_info: Vec<BatchInfo> = match serde_json::from_slice(&body) {
+                            Ok(b) => b,
+                            Err(err) => {
+                                return Either::B(future::err(RestApiResponseError::InternalError(
+                                    format!("Failed to parse response body {}", err),
+                                )))
+                            }
+                        };
+
+                        // If batch status is still pending and the wait time has not yet passed,
+                        // send request again to re-check the batch status
+                        if batches_info
+                            .iter()
+                            .any(|batch_info| match batch_info.status {
+                                BatchStatus::Pending => true,
+                                BatchStatus::Valid(_) => true,
+                                _ => false,
+                            })
+                            && Instant::now().duration_since(start_time) < Duration::from_secs(wait)
+                        {
+                            sleep(Duration::new(1, 0));
+                            Either::A(check_batch_status(
+                                client,
+                                &splinterd_url,
+                                &link,
+                                start_time,
+                                wait,
+                            ))
+                        } else {
+                            Either::B(future::ok(batches_info))
+                        }
+                    }
+                    StatusCode::BAD_REQUEST => {
+                        let body_value: serde_json::Value = match serde_json::from_slice(&body) {
+                            Ok(b) => b,
+                            Err(err) => {
+                                return Either::B(future::err(RestApiResponseError::InternalError(
+                                    format!("Failed to parse response body {}", err),
+                                )))
+                            }
+                        };
+
+                        let message = match body_value.get("message") {
+                            Some(value) => value.as_str().unwrap_or("Request was malformed."),
+                            None => "Request malformed.",
```

Comment:
Is there a reason this is "Request malformed" instead of "Request was malformed."?

Reply:
No reason... I changed them to say the same thing.